packages/liveblocks-core/src/__tests__/_utils.ts

```diff
@@ -184,30 +178,32 @@ export async function prepareRoomWithStorage<
   const effects = mockEffects();
   (effects.send as jest.MockedFunction<any>).mockImplementation(onSend);
 
-  const machine = makeStateMachine<TPresence, TStorage, TUserMeta, TRoomEvent>(
-    makeMachineConfig(effects),
-    {} as TPresence,
-    defaultStorage || ({} as TStorage)
+  const room = createRoom<TPresence, TStorage, TUserMeta, TRoomEvent>(
+    {
+      initialPresence: {} as TPresence,
+      initialStorage: defaultStorage || ({} as TStorage),
+    },
+    makeMachineConfig(effects)
   );
   const ws = new MockWebSocket("");
 
-  machine.connect();
-  machine.authenticationSuccess(makeRoomToken(actor, scopes), ws as any);
+  room.__internal.connect();
+  room.__internal.authenticationSuccess(makeRoomToken(actor, scopes), ws);
   ws.open();
 
   // Start getting the storage, but don't await the promise just yet!
-  const getStoragePromise = machine.getStorage();
+  const getStoragePromise = room.getStorage();
```

Comment:
I think this is done to fire the promise, and then immediately call `room.__internal.onMessage()` on the next line, and only _then_ await the promise. Likely just an optimization? Not sure why it's written like this, this would be my guess.